src/main/java/com/oneops/proxy/clients/proxy/MSProxyClient.java

```diff
@@ -140,13 +140,17 @@ public void setup(String domain) throws GeneralSecurityException {
     ErrorRes err = null;
     T body = null;
 
-    if (res.isSuccessful()) {
-      body = res.body();
+    if (res.isSuccessful() && res.body() != null) {
+      if(((MSClientAuthResponse) res.body()).getAuthorized().get(0).isAuthorized()) {
+        log.info("Passed MS authorization request successfully.");
+        body = res.body();
+        return new Result<>(body, err, res.code(), res.isSuccessful());
```

Comment:
Let's have this logic in the `auth` method. Exec call should only handle the HTTP call execution. The business logic to deal with MSClintAuthResponse should be there in `doAuth(MSClientAuthRequest authRequest)` method.

Reply:
@sureshg : Problem was not in authorization logic. Issue was their while returning the response. When user didn't have auth access, it was only checking "res.isSuccessful()". which was 200 Ok response and having responseBody. This MS service auth response. 
{
    "namespace": "ms-sf-tomcat_tomcat-prodtest_PROD",
    "authorized": [
        {
            "user": "g0k00r4",
            "authorized": false,
            "error": "",
            "metadata": {}
        }
    ]
}

Based on new logic, I have added validation for "authorized" variable.